core/services/ardupilot_manager/mavlink_proxy/Endpoint.py

```diff
@@ -1,76 +1,56 @@
-from dataclasses import dataclass
-from enum import IntEnum
-from typing import Dict, Optional, Union
+from enum import Enum
+from typing import Any, Dict, Type
 
+import validators
+from pydantic import dataclasses, root_validator
 
-class EndpointType(IntEnum):
-    Undefined = 0
-    UDPServer = 1
-    UDPClient = 2
-    TCP = 3
-    Serial = 4
-    File = 5
 
+def is_valid_filepath(filepath: str) -> bool:
+    return filepath.startswith("/") and not filepath.endswith("/")
 
-@dataclass
-class Endpoint:
-    place: str
-    argument: Optional[str]
-    connType: EndpointType = EndpointType.Undefined
-
-    _connTypeMap = [
-        (["udpin"], EndpointType.UDPServer),
-        (["udp", "udpout"], EndpointType.UDPClient),
-        (["tcp"], EndpointType.TCP),
-        (["serial"], EndpointType.Serial),
-        (["file"], EndpointType.File),
-    ]
-
-    def __init__(self, constructor: Union[str, Dict[str, str]] = "") -> None:
-        if isinstance(constructor, dict):
-            self.from_dict(constructor)
-        else:
-            self.from_str(constructor)
-
-    def from_str(self, pattern: str) -> "Endpoint":
-        args = pattern.split(":")
-        if len(args) != 2 and len(args) != 3:
-            raise RuntimeError("Wrong format for endpoint creation.")
-
-        for stringType, endpointType in self._connTypeMap:
-            if args[0] in stringType:
-                self.connType = endpointType
-                break
-        else:
-            self.connType = EndpointType.Undefined
 
-        self.place = args[1]
-        self.argument = args[2] if len(args) == 3 else None
+class EndpointType(str, Enum):
+    UDPServer = "udpin"
+    UDPClient = "udpout"
+    TCP = "tcp"
+    Serial = "serial"
+    File = "file"
 
-        return self
 
-    def from_dict(self, pattern: Dict[str, str]) -> "Endpoint":
-        if not {"connType", "place"}.issubset(pattern.keys()):
-            diff = {"connType", "place"}.difference(pattern.keys())
-            raise RuntimeError(f"Wrong format for endpoint creation, missing the following: {diff}.")
-
-        for stringType, endpointType in self._connTypeMap:
-            if pattern["connType"] in stringType:
-                self.connType = endpointType
-                break
-        else:
-            self.connType = EndpointType.Undefined
-
-        self.place = pattern["place"]
-        self.argument = pattern["argument"] if pattern["argument"] else None
-
-        return self
+@dataclasses.dataclass
+class Endpoint:
+    connection_type: str
+    place: str
+    argument: int
+
+    @root_validator
+    @classmethod
+    def is_mavlink_endpoint(cls: Type["Endpoint"], values: Any) -> Any:
+        connection_type, place, argument = (values.get("connection_type"), values.get("place"), values.get("argument"))
+        if connection_type in [EndpointType.UDPServer, EndpointType.UDPClient, EndpointType.TCP]:
```

Comment:
Can we return the possible options when connection_type has the wrong argument ?

Reply:
Yes, that would be good.